src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/DefaultHttpClient.java

```diff
@@ -76,19 +72,31 @@ public void saveFromResponse(HttpUrl url, List<Cookie> cookies) {
                     cookieLock.unlock();
                 }
             }
-        }).build();
+        }).readTimeout(pollTimeout, TimeUnit.MILLISECONDS)
```

Comment:
Does this affect every request? If so, shouldn't we set the poll timeout for just poll requests and another (probably shorter) timeout for everything else?

Reply:
Yeah, unfortunately in the OkHttp library we have to set the timeout at the client level. 
We'd have to set up 2 HttpClients, one specific for polls and another for all other requests.

Reply:
> We'd have to set up 2 HttpClients, one specific for polls and another for all other requests.

We had to do this in the old C# client to get around some nasty behavior with ServicePoint queuing requests after the poll. Poll requests are special enough that this is a reasonable course of action IMHO.

Reply:
Ah okay. If that doesn't sound unreasonable then we can do that